pkg/blackduck/ctl_blackduck.go

```diff
@@ -340,8 +348,13 @@ func (ctl *Ctl) SetFlag(f *pflag.Flag) {
 			ctl.Spec.AuthCustomCA = data
 		case "type":
 			ctl.Spec.Type = ctl.Type
-		case "desired-state":
-			ctl.Spec.DesiredState = ctl.DesiredState
+		case "start", "stop":
```

Comment:
No need for the complexity here.  We've disallowed both start and stop to be provided together, so just set the spec.DesiredState to START or STOP in separate case statements.  ie

case "start":
  ctl.Spec.DesiredState = "START"
case "stop":
  ctl.Spec.DesiredState = "STOP"